### PR TITLE
Create codeql-analysis.yml

### PR DESCRIPTION
**Pull request by FalseLobster**
### Platforms Impacted
- [ ] iOS
- [ ] macOS
- [x] win32 (Office)
- [x] windows
- [ ] android

### Description of changes

Adds codeql analysis per Microsoft security guidelines

### Verification

This does some analysis at PR time for Security vulnerabilities.  I'm using this PR to validate the change.

### Pull request checklist

This PR has considered (when applicable):
- [ ] Automated Tests
- [ ] Documentation and examples
- [ ] Keyboard Accessibility
- [ ] Voiceover
- [ ] Internationalization and Right-to-left Layouts

### CODE REVIEW

**Review by rurikoaraki — APPROVED** [review]

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,70 @@
+# For most projects, this workflow file will not need changing; you simply need
```
Comment (Saadnajmi) [inline .github/workflows/codeql-analysis.yml:1]: Maybe add a comment where this is coming from?

Comment (FalseLobster) [reply .github/workflows/codeql-analysis.yml:1]: It was autogenerated by GitHub.  I turned it on from the Repository settings

File: .github/workflows/codeql-analysis.yml [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+      actions: read
+      contents: read
+      security-events: write
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [ 'cpp', 'javascript' ]
+        # CodeQL supports [ 'cpp', 'csharp', 'go', 'java', 'javascript', 'python', 'ruby' ]
+        # Learn more about CodeQL language support at https://git.io/codeql-language-support
```
Comment (Saadnajmi) [inline .github/workflows/codeql-analysis.yml:37]: The link seems to suggest TypeScript is also supported?
Also, sad no Objective C or Swift support :(

Comment (FalseLobster) [reply .github/workflows/codeql-analysis.yml:37]: Typescript appears to be included in the javascript analysis

**Review by Saadnajmi — APPROVED** [review]

**Comment by rurikoaraki** [comment]
@FalseLobster is there an item/issue logged for turning the cpp check back on?

**Comment by FalseLobster** [comment]
> @FalseLobster is there an item/issue logged for turning the cpp check back on?

Thanks for keeping me accountable, I opened #1416 

I just figured it'd take me longer to figure out how to enable the CPP workflow than it's worth right now considering this repo is 99% TS.  Having CodeQL coverage for that seems more pressing.

**Review by rurikoaraki — APPROVED** [review]

**Pull request merged**